galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -353,3 +358,94 @@ def _remove_content(collection_version, repo):
         locks = [repo]
         task_args = (collection_version.pk, repo.pk)
         return enqueue_with_reservation(remove_content_from_repository, locks, args=task_args)
+
+
+class CollectionVersionDependencyViewSet(api_base.LocalSettingsMixin,
+                                         pulp_ansible_views.CollectionVersionViewSet):
+    model = CollectionVersion
+    serializer_class = CollectionVersionDependencySerializer
+    permission_classes = [access_policy.CollectionAccessPolicy]
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Returns the resolved Collection dependencies of the specified CollectionVersion.
+        """
+        serializer = self.get_serializer_class()(
+            self.get_object(),
+            context=self.get_serializer_context()
+        )
+
+        dependency_map = {}
+        resolved_deps = {}
+
+        # Initial dependencies
+        deps = serializer.data['dependencies']
+
+        while len(deps) > 0:
+            resolved_version_number = ''
+            new_dependencies = {}
+            current_dep = ''
+
+            for key in deps:
+                current_dep = key
+
+                # Add dependencies to dependency map
+                if key in resolved_deps and key in dependency_map:
+                    resolved_deps.pop(key)
+                    dependency_map[key].append(deps[key])
+                    deps[key] += ',' + ','.join(dependency_map[key])
+                else:
+                    if key in dependency_map:
+                        dependency_map[key].append(deps[key])
+                        deps[key] = ','.join(dependency_map[key])
+                    else:
+                        dependency_map[key] = [deps[key]]
+
+                # Build then run sql query based on deps to get the latest compatible version
+                namespace, name = key.split('.')
+                split_deps = deps[key].split(',')
+                filters = []
+                for d in split_deps:
+                    offset = 0
+                    if '>=' in d or '<=' in d:
+                        offset = 2
+                    if '=' in d or '>' in d or '<' in d:
+                        offset = 1
+                    operand = d[:offset]
+                    version = d[offset:]
+                    if '>=' == operand:
```

Comment:
You can use the "dispatch dictionary" pattern to replace this if-elif-else construction.